### PR TITLE
Conversion msh > xdmf/h5 for mesh, subdomains, interfaces

### PR DESCRIPTION
**Pull request by anzil**
Introduced a separate mesh file conversion method. And adapted the `bingham` example accordingly.

### CODE REVIEW

**Comment by michalhabera** [comment]
@anzil Can you check the new meshio api (e.g. in updated bingham annulus mesh) and update this PR?

**Comment by anzil** [comment]
The new meshio API is now reflected in the last commit to this branch.

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,25 @@
+FROM quay.io/fenicsproject/dolfinx:dev-env-real
```
Comment (michalhabera) [inline Dockerfile]: Why is this needed? We have similar Dockerfile here https://github.com/FEniCS/dolfinx/blob/master/docker/Dockerfile

**Comment by michalhabera** [comment]
My latests commits are weirdly not processed by circleCI, I need to figure out why.

**Comment by michalhabera** [comment]
Ok, circleCI is again running.

File: Dockerfile (hunk 1 above)
Comment (anzil) [reply Dockerfile]: To have this (much shorter) `Dockerfile` is primarily for convenience, secondly a must-have for customised test env containers in the scenario that certain dolfin-x features are not (yet) in master.

Comment (michalhabera) [reply Dockerfile]: For such case you could just pull dev-env-real and build dolfinx inside the container. We can keep this bit here, but need to remember this code is not tested.

**Comment by michalhabera** [comment]
Thanks @anzil !
